x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -160,83 +164,102 @@ export function App({
-            <nav>
-              <EuiFlexGroup>
-                <EuiFlexItem grow={false}>
-                  <EuiLink
-                    data-test-subj="lnsApp_saveButton"
-                    onClick={() => {
-                      if (isSaveable && lastKnownDocRef.current) {
-                        docStorage
-                          .save(lastKnownDocRef.current)
-                          .then(({ id }) => {
-                            // Prevents unnecessary network request and disables save button
-                            const newDoc = { ...lastKnownDocRef.current!, id };
-                            setState({
-                              ...state,
-                              isDirty: false,
-                              persistedDoc: newDoc,
-                            });
-                            if (docId !== id) {
-                              redirectTo(id);
-                            }
-                          })
-                          .catch(reason => {
-                            core.notifications.toasts.addDanger(
-                              i18n.translate('xpack.lens.editorFrame.docSavingError', {
-                                defaultMessage: 'Error saving document {reason}',
-                                values: { reason },
-                              })
-                            );
-                          });
-                      }
-                    }}
-                    color={isSaveable ? 'primary' : 'subdued'}
-                    disabled={!isSaveable}
-                  >
-                    {i18n.translate('xpack.lens.editorFrame.save', {
-                      defaultMessage: 'Save',
-                    })}
-                  </EuiLink>
-                </EuiFlexItem>
-              </EuiFlexGroup>
-            </nav>
-            <QueryBarTopRow
-              data-test-subj="lnsApp_queryBar"
+            <TopNavMenu
+              config={[
+                {
+                  label: i18n.translate('xpack.lens.editorFrame.save', {
+                    defaultMessage: 'Save',
+                  }),
+                  run: () => {
+                    if (isSaveable && lastKnownDocRef.current) {
+                      docStorage
+                        .save(lastKnownDocRef.current)
+                        .then(({ id }) => {
+                          // Prevents unnecessary network request and disables save button
+                          const newDoc = { ...lastKnownDocRef.current!, id };
+                          setState(s => ({
+                            ...s,
+                            isDirty: false,
+                            persistedDoc: newDoc,
+                          }));
+                          if (docId !== id) {
+                            redirectTo(id);
+                          }
+                        })
+                        .catch(reason => {
+                          core.notifications.toasts.addDanger(
```

Comment:
You can use `addError` instead, which takes the error object directly. I think that would probably be preferable here.